addons/web_editor/static/src/js/editor/image_processing.js

```diff
@@ -263,12 +263,29 @@ async function applyModifications(img, dataOptions = {}) {
  * @param {String} src URL of the image to load
  * @param {HTMLImageElement} [img] img element in which to load the image
  * @returns {Promise<HTMLImageElement>} Promise that resolves to the loaded img
+ *     or a placeholder image if the src is not found.
  */
 function loadImage(src, img = new Image()) {
-    return new Promise((resolve, reject) => {
-        img.addEventListener('load', () => resolve(img), {once: true});
-        img.addEventListener('error', reject, {once: true});
+    const handleImage = (src, resolve, reject) => {
```

Comment:
Seems confusing to add a new param `src` which is shadowing the one from `loadImage`:
1. Did you really need to add it?
2. If yes, can't you name it otherwise so there is no confusion/collision?